stdlib/domain.mli

```diff
@@ -16,6 +16,12 @@
 (*                                                                        *)
 (**************************************************************************)
 
+[@@@alert unstable
+    "The Domain module is experimental in OCaml 5.0.\
+     Its interface might be refined in future versions and \
+     break forward compatibility."
```

Comment:
Since the alert is now `unstable` (which I think is better, it seems that's what other languages use aswell). I would suggest to drop the experimental terminology. I also wouldn't mention the version number. Besides I personally always get confused about forward and backward compatibility because the point where the person speaks from and where you read from makes it a bit unclear.

I would propose one of these formulations (repeating `unstable` depends a bit on how alerts render in the docs and in the warnings which I'm unclear of right now):

```
The [Domain] interface may change in the future.
The [Domain] interface is unstable, it may change in the future.
The [Domain] interface may change in incompatible ways in the future.
The [Domain] interface is unstable, it may change in incompatible ways in the future.
```

Reply:
It is generally useful to err on the stricter side in the compiler code base itself. For instance, we want to make it harder to introduce surreptitiously effects and effect handlers in the typechecker.

Good point on the alert message, I was blind-sided by the mechanical side of the update. The alert is currently rendered as

> **Alert unstable.** ...

in the documentation and
> **Alert** unstable: module Stdlib.Domain
> ...
in error messages.

Thus

> **Alert unstable.** The [Domain] interface may change in incompatible ways in the future.

seems like a good option to me.